contracts/cw-core/src/contract.rs

```diff
@@ -728,3 +786,60 @@ pub fn reply(deps: DepsMut, _env: Env, msg: Reply) -> Result<Response, ContractE
         _ => Err(ContractError::UnknownReplyID {}),
     }
 }
+
+// Convert to base26, where we only use the characters A-Z. Note that since 'A' maps to 0,
+// a prefix will never start with 'A' unless it is 0.
+pub(crate) fn derive_proposal_module_prefix(mut dividend: usize) -> StdResult<String> {
+    // Pre-allocate string
+    let mut prefix = String::with_capacity(10);
+    loop {
+        let remainder = dividend % 26;
+        dividend /= 26;
+        let remainder_str = std::str::from_utf8(&[(remainder + 65) as u8])?.to_owned();
+        prefix.push_str(&remainder_str);
+        if dividend == 0 {
+            break;
+        }
+    }
+    Ok(prefix.chars().rev().collect())
+}
```

Comment:
I think we want `26` to yield `AA`, `27` to yield `AB`, and `28` to yield `AC`, increasing in that pattern. This algorithm doesn't seem to ever output `AA`, as 25 is `Z` and 26 is `BA`.

Here is the TypeScript algorithm we found online, converted to Rust:

```suggestion
pub(crate) fn derive_proposal_module_prefix(mut dividend: usize) -> StdResult<String> {
    dividend += 1;
    // Pre-allocate string
    let mut prefix = String::with_capacity(10);
    loop {
        let remainder = (dividend - 1) % 26;
        dividend = (dividend - remainder) / 26;
        let remainder_str = std::str::from_utf8(&[(remainder + 65) as u8])?.to_owned();
        prefix.insert_str(0, &remainder_str);
        if dividend == 0 {
            break;
        }
    }
    Ok(prefix.chars().rev().collect())
}
```

Would that work? Or did you modify the algorithm for any reason?

Reply:
yeah this works better! mine was slightly different because I was converting to base26 where all of the "numbers" are A-Z, and since A maps to 0, a prefix wouldn't start with A unless it is 0. but agreed its better to have 27 be AA